apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-modal/index.tsx

```diff
@@ -32,11 +33,25 @@ const LaunchModal: React.FunctionComponent< Props > = ( { onClose } ) => {
 	const [ isLaunching, setIsLaunching ] = React.useState( false );
 
 	const { launchSite } = useDispatch( SITE_STORE );
+	const { savePost } = useDispatch( 'core/editor' );
 
-	const handleLaunch = () => {
+	const handleLaunch = React.useCallback( () => {
 		launchSite( siteId );
 		setIsLaunching( true );
-	};
+	}, [ launchSite, setIsLaunching, siteId ] );
+
+	React.useEffect( () => {
+		if ( ! isLaunching && isLaunchImmediately ) {
+			setIsLaunching( true );
+			savePost();
+
+			// Delay handleLaunch() to allow for the save to complete;
```

Comment:
😞

Reply:
This seems...less than ideal. I'm not super familiar with @wordpress/data stuff, but is there some sort of promise-based way to await the result of saving the post?

https://developer.wordpress.org/block-editor/packages/packages-data/ I haven't looked into it in depth but the `subscriber` section of this documentation seems like it could be promising.

Reply:
See example elsewhere:

https://github.com/Automattic/wp-calypso/blob/61acc957638fc7009632af4c0c058aa6bc0e82b7/apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts#L79

Reply:
> See example elsewhere:
> 
> https://github.com/Automattic/wp-calypso/blob/61acc957638fc7009632af4c0c058aa6bc0e82b7/apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts#L79
> 
> Is that not working?

Hey thanks, that worked. Done here: 93cf373d36ba. I believe I tried something like this early on, but I was scared away from a warning displayed in my editor:  `80007: 'await' has no effect on the type of this expression.`. It shows that when I look at the example linked above, too (`wp-calypso/apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts`, Line 79 in `61acc95`)

However, given that
* I can't get `eslint` or `tsc` to show those warnings on the command line
* It works in my testing

I'm going to consider it to be either a misconfigured editor or over-enthusiastic warning.  I see the warning in both vscode and vim + vim-coc.